flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/StringUtils.java

```diff
@@ -0,0 +1,454 @@
+                for (int i = 0; i < array.length; i++) {
+                    formattedArray[i] = toSQLString(array[i], elementType, sessionTimeZone);
+                }
+                return Arrays.toString(formattedArray);
+            } else if (value.getClass() == float[].class) {
+                float[] array = (float[]) value;
+                String[] formattedArray = new String[array.length];
+                for (int i = 0; i < array.length; i++) {
+                    formattedArray[i] = toSQLString(array[i], elementType, sessionTimeZone);
+                }
+                return Arrays.toString(formattedArray);
+            } else if (value.getClass() == double[].class) {
+                double[] array = (double[]) value;
+                String[] formattedArray = new String[array.length];
+                for (int i = 0; i < array.length; i++) {
+                    formattedArray[i] = toSQLString(array[i], elementType, sessionTimeZone);
+                }
+                return Arrays.toString(formattedArray);
+            } else if (value.getClass() == boolean[].class) {
+                boolean[] array = (boolean[]) value;
+                String[] formattedArray = new String[array.length];
+                for (int i = 0; i < array.length; i++) {
+                    formattedArray[i] = toSQLString(array[i], elementType, sessionTimeZone);
+                }
+                return Arrays.toString(formattedArray);
+            } else if (value.getClass() == char[].class) {
+                char[] array = (char[]) value;
+                String[] formattedArray = new String[array.length];
+                for (int i = 0; i < array.length; i++) {
+                    formattedArray[i] = toSQLString(array[i], elementType, sessionTimeZone);
+                }
+                return Arrays.toString(formattedArray);
+            } else if (value.getClass().isArray()) {
+                // non-primitive type
+                Object[] array = (Object[]) value;
+                String[] formattedArray = new String[array.length];
+                for (int i = 0; i < array.length; i++) {
+                    formattedArray[i] = toSQLString(array[i], elementType, sessionTimeZone);
+                }
+                return Arrays.toString(formattedArray);
+            }
+            return Objects.toString(value);
+        }
+
+        @Override
+        public String visit(RowType rowType) {
+            if (value instanceof Row) {
+                Row row = (Row) value;
+                String[] formattedArray = new String[row.getArity()];
+                for (int i = 0; i < rowType.getFields().size(); i++) {
+                    LogicalType type = rowType.getFields().get(i).getType();
+                    formattedArray[i] = toSQLString(row.getField(i), type, sessionTimeZone);
+                }
+                return "(" + String.join(", ", formattedArray) + ")";
```

Comment:
This `Row` string representation is the same we have in `ScalarOperatorGens#generateCastRowDataToString`